Feature/use mobius tokio

saturn77 opened the pull request:
# Update planner_gui_egui to use local Tokio runtime

This PR updates planner_gui_egui to use a local Tokio runtime implementation with egui_mobius Signal/Slot handling.
  
## Changes

## Primary:
- Added `runtime.rs` with a local `TokioRuntime` wrapper 
- Provides clean thread management and runtime access
- Task system with future handling
- Added into_future function with type bounds and boxing
- Fixed stream-to-future conversion for async tasks

## Secondary: 
- Updated egui_mobius api calls to be congruent to crates.io release 0.3.0-alpha.17 version of egui_mobius
- Updated `ui_component.rs` to use new egui_mobius signal-slot API without name id 
- Removed unnecessary buffer size parameter from `create_signal_slot`reflecting updated id 
- Simplified signal-slot handling in UiApp by removing MaybeUninit wrapper
- Fixed proper creation of signal-slot pairs in both Default and new()

## Testing
- Application launches and handles async operations correctly (Verified compilation and basic runtime functionality)

hydra commented on crates/planner_gui_egui/src/ui_app.rs:
```diff
@@ -241,8 +239,8 @@ impl UiApp {
-                if let Some(stream) = task::into_stream(task) {
-                    runtime.run(stream);
+                if let Some(future) = crate::task::into_future(task) {
+                    runtime.runtime().spawn(future);
```
this no-longer dispatches the ui commands that the future returns.  See the deleted `MessageDispatcher`.

this works:
```rust
if let Some(mut stream) = task::into_stream(task) {
    runtime.runtime().spawn({
        let app_message_sender = app_message_sender.clone();
        async move {
            debug!("running stream future");
            while let Some(command) = stream.next().await {
                debug!("command returned from future: {:?}", command);
                app_message_sender.send(command).expect("sent");
            }
        }
    });
}
```

hydra commented:
I fixed some of the issues outlined in my changes above, see these commits:

https://github.com/MakerPnP/makerpnp/commits/feature/use-mobius-tokio/

Can you pull those commits into your PR branch and resolve the remaining items.

saturn77 commented on crates/planner_gui_egui/src/main.rs:
```diff
@@ -1,37 +1,21 @@
-use tracing_subscriber::{EnvFilter, fmt};
 
 fn main() {
-    tracing_subscriber::registry()
```
adding it back in, sorry about that, there were a series of changes with tokio and not sure how that happened!

saturn77 commented on crates/planner_gui_egui/src/main.rs:
```diff
@@ -1,37 +1,21 @@
 use i18n::I18nConfig;
 use planner_gui_egui::ui_app::UiApp;
-/// Run as follows:
-/// `run --package planner_gui_egui --bin planner_gui_egui`
```
adding it back in along with above on the tracing_subscriber

saturn77 commented on Cargo.toml:
```diff
@@ -19,6 +19,7 @@ members = [
     "crates/criteria",
     "crates/planner_app",
     "crates/planner_cli",
+    "crates/makerpnp_reactive",
```
will leave off reactive as well as the widgets crate

saturn77 commented:
> I fixed some of the issues outlined in my changes above, see these commits:
> 
> https://github.com/MakerPnP/makerpnp/commits/feature/use-mobius-tokio/
> 
> Can you pull those commits into your PR branch and resolve the remaining items.

Have pushed the latest commits to branch "reactive-runtime" 

## Basics
1. removed egui_reactive
2. bumped to latest egui_mobius stable, 0.3.0-alpha.23 

## Primary
1. Merged your feature/use-mobius-tokio into reactive_runtime
2. Currently compiles and runs , but did not load a planner file for testing at that level

hydra commented:
| Have pushed the latest commits to branch "reactive-runtime"

The changes should be in the branch used to create this PR.

hydra commented:
I amended your commits and rebased on master and have merged it.

notable changes:
* restored main.rs
* restored runtime/mod.rs and moved the code to runtime/legacy_runtime.rs
* move runtime.rs to runtime/tokio_runtime.rs

I'm not ready to delete the old runtime yet since it still contains tests that have not been ported to the tokio runtime.
